src/wagtail_live/static/wagtail_live/js/websocket/websocket.js

```diff
@@ -0,0 +1,60 @@
+class WebsocketPublisher {
+    /**
+     * Initializes and pens a new websocket connection with server side 
+     * or a real-time message service.
+     * @constructor
+     * @param {String} baseURL - baseURL to use for websocket connections.
+     */
+    constructor(baseURL) {
+        this.baseURL = baseURL;
+        this.initialize_websocket_connection();
+    }
+
+    /**
+     * Registers callbacks on websockets events.
+     */
+    start() {
+        this.initialize_on_message_event();
+        this.initialize_on_error_event();
+    }
+
+    /**
+     * Opens a websocket connection with the baseURL provided.
+     */
+    initialize_websocket_connection() {}
+
+    /**
+     * Registers callback on new message events.
+     */
+    initialize_on_message_event() {}
+
+    /**
+     * Registers callback on disconnect/error events.
+     */
+    initialize_on_error_event() {}
+
+}
+
+class GenericWebsocketPublisher extends WebsocketPublisher {
+    initialize_websocket_connection() {
+        this.websocket = new WebSocket(
+            'ws://'
+            + this.baseURL
+            + '/ws/channel/'
```

Comment:
Is this notation preferable to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals?

Reply:
I will use the template literals.